planner/cascades/transformation_rules.go

```diff
@@ -1254,6 +1258,102 @@ func (r *MergeAggregationProjection) OnTransform(old *memo.ExprIter) (newExprs [
 	return []*memo.GroupExpr{newAggExpr}, true, false, nil
 }
 
+// EliminateSingleMaxMin tries to convert a single max/min to Limit+Sort operators.
+type EliminateSingleMaxMin struct {
+	baseRule
+}
+
+// NewRuleEliminateSingleMaxMin creates a new Transformation EliminateSingleMaxMin.
+// The pattern of this rule is `max/min->X`.
+func NewRuleEliminateSingleMaxMin() Transformation {
+	rule := &EliminateSingleMaxMin{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandAggregation,
+		memo.EngineTiDBOnly,
+		memo.NewPattern(memo.OperandAny, memo.EngineTiDBOnly),
+	)
+	return rule
+}
+
+// Match implements Transformation interface.
+func (r *EliminateSingleMaxMin) Match(expr *memo.ExprIter) bool {
+	// Use appliedRuleSet in GroupExpr to avoid re-apply rules.
+	if expr.GetExpr().HasAppliedRule(r) {
+		return false
+	}
+
+	agg := expr.GetExpr().ExprNode.(*plannercore.LogicalAggregation)
+	// EliminateSingleMaxMin only works on the complete mode.
+	if !agg.IsCompleteModeAgg() {
+		return false
+	}
+	if len(agg.GroupByItems) != 0 {
+		return false
+	}
+	if len(agg.AggFuncs) != 1 {
```

Comment:
This check is not the best. Consider we have the aggregation functions: `max(a), first_row(b)`, we can also eliminate max.

Reply:
I have no good idea to deal with this situation? In the origin [implementation](https://github.com/pingcap/tidb/blob/master/planner/core/rule_max_min_eliminate.go#L210-L214), it only checks this. Do you have some ideas? @lzmhhh123  @francis0407

Reply:
Maybe the condition should be:
1. only one `max()` or `min()` in the `AggFuncs` slice, and:
2. the other aggregate functions in the `AggFuncs` slice should be `FirstRow()`.

Since we are refactoring a brand new cascades planner, we'd better try our best to do better than the old one. 😄

Reply:
+1.
And also add a test case like `select max(a), first_row(b) from t`.

Reply:
Here is a simple example: 
```sql
select max(a), b from t group by b;
```

Reply:
@lzmhhh123 Can `Agg` contain `GroupByItems`? I think this optimization only works when Agg has no `GroupByItem`.